<launch>
	<arg name="veh" default="$(env VEHICLE_NAME)"/>
	<arg name="ai_trafo_mode" default="cb" doc="'cb' for colo balance only; 'both' for color balance and linear trafo"/>
	<arg name="ai_interval" default="5" doc="interval with which the linear trafo gets updated. color balance is performed every second."/>
	<arg name="verbose" default="false"/>
	<arg name="pure_pursuit" default="true" />
	<arg name="pkg_name" default="pure_pursuit" doc="name of the package"/>
	<arg name="node_name" default="pure_pursuit_node" doc="name of the node"/>
	<arg name="param_file_name" default="default" doc="Specify a param file. ex:megaman"/>

	<!-- start basic args -->
	<include file="$(find duckietown_demos)/launch/master.launch">
		<arg name="veh" value="$(arg veh)"/>
		<arg name="visualization" default="true" />
		<!--arg name="verbose" value="$(arg verbose)" /-->
    	<arg name="/camera/raw" default="true" />
		<arg name="param_file_name" default="default" />
		<arg name="line_detector_param_file_name" default="default" />
		<arg name="anti_instagram" default="true" />
		<arg name="ai_trafo_mode" value="$(arg ai_trafo_mode)"/>
		<arg name="ai_interval" value="$(arg ai_interval)"/>
		<arg name="/lane_following/lane_controller" default="false" />
		<arg name="vehicle_avoidance" default="false"/>
	</include>

	<group ns="$(arg veh)">
		<remap from="car_cmd_switch_node/cmd_lane_following" to="pure_pursuit_node/lane_control"/>
		<remap from="pure_pursuit_node/lane_pose" to="lane_filter_node/lane_pose"/>
		<remap from="pure_pursuit_node/seglist_filtered" to="lane_filter_node/seglist_filtered"/>
		<!--NO!!! <remap from="fsm_node/mode" to="fsm_node/lane_pose_intersection_navigation"/> --><!--TODO what is this -->
		<!-- <remap from="pure_pursuit_node/lane_pose_obstacle_avoidance" to="lane_filter_node/lane_pose_obstacle_avoidance"/> TODO: add correct node_name
		<remap from="pure_pursuit_node/lane_pose_parking" to="lane_filter_node/lane_pose_parking"/>
		<remap from="pure_pursuit_node/implicit_coordination_velocity" to="lane_filter_node/implicit_coordination_velocity"/> -->
		<!--<remap from="pure_pursuit_node/wheels_cmd_executed" to="wheels_driver_node/wheels_cmd_executed" />-->
		<remap from="pure_pursuit_node/fsm_mode" to="fsm_node/mode" />
		<!--<remap from="pure_pursuit_node/obstacle_avoidance_pose" to="obst_avoid/obstacle_avoidance_pose" />
		<remap from="pure_pursuit_node/obstacle_detected" to="obstacle_avoidance_node/obstacle_avoidance_active_flag" />
		<remap from="pure_pursuit_node/stop_line_reading" to="stop_line_filter_node/stop_line_reading" />-->
		<remap from="wheels_driver_node/radius_limit" to="pure_pursuit_node/radius_limit" />
		<!--<remap from="vehicle_avoidance_control_node/car_cmd" to="pure_pursuit_node/car_cmd" />
		<remap from="vehicle_avoidance_control_node/car_cmd_in" to="pure_pursuit_node/car_cmd"/>-->
		<node  pkg="$(arg pkg_name)" type="$(arg node_name)_slim.py" name="$(arg node_name)" output="screen" required="true">
	    </node>
	</group>
</launch>
